### PR TITLE
Migrate from javax.ws.rs to jakarta.ws.rs

### LINKED ISSUE #259: Migrate from javax.ws.rs to jakarta.ws.rs
### Issue description

Migrate from: 

https://search.maven.org/artifact/org.jboss.resteasy/jaxrs-api/3.0.12.Final/jar

to: 
https://search.maven.org/artifact/jakarta.ws.rs/jakarta.ws.rs-api/3.1.0/jar

### PR DESCRIPTION
**Pull request by wetted**
closes #259

### CODE REVIEW

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsRuntime.java [hunk 1]
```diff
@@ -40,12 +38,6 @@ public class JaxRsRuntime {
      * @param application The application
      */
     protected JaxRsRuntime(ApplicationContext applicationContext, Application application) {
-        final Set<Object> singletons = application.getSingletons();
```
Comment (sdelamo) [inline jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsRuntime.java:43]: why did you remove this?

**Review by sdelamo — CHANGES_REQUESTED** [review]
We have to add a `breaks.adoc` section. We need to tell the users that we have migrated to jakarta namespace.

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsRuntime.java (hunk 1 above)
Comment (wetted) [reply jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsRuntime.java:43]: `Application.getSingletons()` in core is deprecated, and all it does is return `Collections.emptySet()`, so there is nothing to register.

**Comment by wetted** [comment]
The latest commit needed some of the same changes in #266 to avoid compile failures. 

`SimpleSecurityContextTest` is broken until this module points to the 4.x version of `micronaut-security` (for the `ConversionService` refactor).

I'm not yet sure what's going on with `HeadersTest`. This wasn't broken before.

**Comment by sdelamo** [comment]
@timyates can you review this PR and address anything necessary to merge into master.

**Comment by timyates** [comment]
@sdelamo should be good now...  apart from the coverage...

**Review by timyates — APPROVED** [review]
One question for @sdelamo and @graemerocher 

Do we want to keep the original annotation mappers, so that people using javax.ws.* annotations can continue to do so?

Or are we good with dropping support for them in 4.0.0?

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
> One question for @sdelamo and @graemerocher
> 
> Do we want to keep the original annotation mappers, so that people using javax.ws.* annotations can continue to do so?
> 
> Or are we good with dropping support for them in 4.0.0?

can you add a paragraph in `breaks.adoc` describing we dropped them and they have to use jakarta.
